### ISSUE TITLE
Decrease the number of suggestions printed on failure

### ISSUE THREAD

**Issue by hvelarde** [plone/plone.api#329]
some API methods, like [get_registry_record](https://github.com/plone/plone.api/blob/master/src/plone/api/portal.py#L253-L309) or [get_view](https://github.com/plone/plone.api/blob/master/src/plone/api/content.py#L500-L535) return a large number (sometimes dozens) of possible options in case the requested object is not found.

Would it be a desired behavior to filter that to show only the most similar ones?

**Comment by gforcada** [comment]
It would prefer to keep it the way it is unless we have some really clever algorithm here, as you may have a typo at the beginning, end, in the middle, etc and then the suggestions can be not so good. And anyway one does not want to put code that can raise this exception on production...

I'm slightly -0.5 here, others?

**Comment by hvelarde** [comment]
yes, we have such option: is part of the standard Python library and is fully configurable:

https://docs.python.org/2/library/difflib.html#difflib.get_close_matches

**Comment by jaroel** [comment]
@hvelarde do have examples of both? Then I can check if using difflib would improve things.
I think it would for get_registry_record:
ipdb> difflib.get_close_matches('pone.app.querstring.itle', records)
['plone.app.querystring.field.end.title', 'plone.app.querystring.field.path.title', 'plone.app.querystring.field.start.title']

I haven't looked at get_view. Maybe get_view should still dump all available views, but do a suggestion first.

@gforcada FYI the exception is still raised, just the message is changed a bit.

**Comment by hvelarde** [comment]
@jaroel here you go: http://pastie.org/10957635
